miden-lib/src/wallets/mod.rs

```diff
@@ -0,0 +1,60 @@
+use crate::assembler::assembler;
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    crypto::merkle::MerkleStore,
+    utils::{format, string::String, vec},
+    AccountError, Word, ZERO,
+};
+
+use vm_core::crypto::dsa::rpo_falcon512::PublicKey;
+
+pub enum AuthScheme {
+    RpoFalcon512 { pub_key: PublicKey }, // we'll later change this to `PublicKey` from Falcon module.
+}
+
+/// Creating accounts with the basic wallet code, the authentication scheme can be defined.
+pub fn create_basic_wallet(
+    init_seed: [u8; 32],
+    auth_scheme: AuthScheme,
+) -> Result<(Account, Word), AccountError> {
+    let (auth_scheme_procedure, storage_slot_0): (&str, Word) = match auth_scheme {
+        AuthScheme::RpoFalcon512 { pub_key } => ("basic::auth_tx_rpo_falcon512", pub_key.into()),
+    };
+
+    let account_code_string: String = format!(
+        "
+    use.miden::wallets::basic->basic_wallet
+    use.miden::eoa::basic
```

Comment:
minor nit: for consistency, should we alias this as `basic_eoa` as well?

Reply:
I thought about it, because we had it before. But in the future ppl. might want to use non-basic authentication schemes, so `basic` should be part of the variable

Reply:
Makes sense. Let's keep it as is.